### ISSUE TITLE
Multicriteria search - bug in TranslationSearch.php on line 235

### ISSUE THREAD

**Issue by pchamo** [OCSInventory-NG/OCSInventory-ocsreports#647]
### General informations
Operating system : 

### Server installation method ( Only one choice )
- [ ] RPM Package
- [ ] Install SH
- [X] Manual installation (OCSNG_UNIX_SERVER_2.6_RC.tar.gz)

### Server informations
Php version : php-5.4.16-46.el7.x86_64
Mysql / Mariadb / Percona version : mysql-community-server-5.7.24-1.el7.x86_64
Apache version : httpd-2.4.6-88.el7.centos.x86_64

### OCS Inventory informations
Ocsreports version : 

### Problem's description

If fails when clicking on multicriteria searches, it shows only a blank page. Reviewing the Apache logs, the error is:

> PHP Fatal error:  Can't use method return value in write context in /usr/share/ocsinventory-reports/ocsreports/require/search/TranslationSearch.php on line 235, referer: https://ovmgr.sgc.gov.co/ocsreports/index.php?first

The offending code is:

```
    public function getTranslationFor($name){
        global $l;
        $name = strtolower($name);
        if(empty($l->g($this->translationArray[$name]))){
            return $name;
        }
        return $l->g($this->translationArray[$name]);
    }

    public function getTranslationForOperator($name){
        global $l;
        return $l->g($this->operatorsArray[$name]);
    }

    public function getTranslationForComparator($name){
        global $l;
        return $l->g($this->comparatorsArray[$name]);
    }

    public function getTranslationForListField($string){
        global $l;

        $values = explode(".", $string);

        if(!empty($l->g($this->translationArray[$values[0]]))){
            $table = $l->g($this->translationArray[$values[0]]);
        }else{
            $table = $values[0];
        }

        $name = strtolower($values[1]);

        if(!empty($l->g($this->translationArray[$name]))){
            $name = $l->g($this->translationArray[$name]);
        }elseif(strpos($name, 'fields_') !== false){
            $accountInfoSearch = new AccountinfoSearch();
            $translateAccount = $accountInfoSearch->getAccountInfosList();
            $name = $translateAccount['COMPUTERS'][$name];
        }

        return $table." : ".$name;

    }
```

I'm not a programmer but this code solves the problem for me:

```
    public function getTranslationFor($name){
        global $l;
        $name = strtolower($name);
        /* pch */
        $vacio=($l->g($this->translationArray[$name]));
        if(empty($vacio)){
/*        if(($l->g($this->translationArray[$name]))){ */
            return $name;
        }
        return $l->g($this->translationArray[$name]);
    }

    public function getTranslationForOperator($name){
        global $l;
        return $l->g($this->operatorsArray[$name]);
    }

    public function getTranslationForComparator($name){
        global $l;
        return $l->g($this->comparatorsArray[$name]);
    }

    public function getTranslationForListField($string){
        global $l;

        $values = explode(".", $string);

        /* pch */
        $vacio=($l->g($this->translationArray[$values[0]]));
        if(!empty($vacio)){

/*        if(!empty($l->g($this->translationArray[$values[0]]))){*/
            $table = $l->g($this->translationArray[$values[0]]);
        }else{
            $table = $values[0];
        }

        $name = strtolower($values[1]);

        /* pch */
        $vacio=($l->g($this->translationArray[$name]));
        if(!empty($vacio)){
/*        if(!empty($l->g($this->translationArray[$name]))){*/
            $name = $l->g($this->translationArray[$name]);
        }elseif(strpos($name, 'fields_') !== false){
            $accountInfoSearch = new AccountinfoSearch();
            $translateAccount = $accountInfoSearch->getAccountInfosList();
            $name = $translateAccount['COMPUTERS'][$name];
        }

        return $table." : ".$name;

    }
```

Differences:

```
[root@ocsserver ~]# diff /root/TranslationSearch.php /usr/share/ocsinventory-reports/ocsreports/require/search/TranslationSearch.php
235,238c235
<         /* pch (my initials, to mark changed lines) */
<         $vacio=($l->g($this->translationArray[$name]));
<         if(empty($vacio)){
< /*        if(($l->g($this->translationArray[$name]))){ */
---
>         if(empty($l->g($this->translationArray[$name]))){
259,263c256
<         /* pch */
<         $vacio=($l->g($this->translationArray[$values[0]]));
<         if(!empty($vacio)){
<
< /*        if(!empty($l->g($this->translationArray[$values[0]]))){*/
---
>         if(!empty($l->g($this->translationArray[$values[0]]))){
271,274c264
<         /* pch */
<         $vacio=($l->g($this->translationArray[$name]));
<         if(!empty($vacio)){
< /*        if(!empty($l->g($this->translationArray[$name]))){*/
---
>         if(!empty($l->g($this->translationArray[$name]))){
```

Vimdiff:

http://prntscr.com/m080oe

Thank you :)

**Comment by gillesdubois** [comment]
Hi,

Update your php to 5.5 and newer will also fix the problem.
5.4 is becoming quite old.

Regards,
Gilles Dubois.

**Comment by pchamo** [comment]
That's ok but the ocs ./setup.sh installer should warn about not having php 7. Thank you.

**Comment by gillesdubois** [comment]
Hi,

A fix has been added for 5.5 compatibilities.
However, 5.4 compatibility. 

Regards,
Gilles Dubois.
